app/models/piece.rb

```diff
@@ -1,3 +1,43 @@
 class Piece < ApplicationRecord
   belongs_to :game
+
+  def obstructed?(destination_x, destination_y)
+    return true if horizontal_or_vertical_obstruction?(destination_x, destination_y)
+    return true if diagonal_obstruction?(destination_x, destination_y)
+    return true if invalid?(destination_x, destination_y)
+
+    false
```

Comment:
Invalid cases would be when the coordinates are off the board (for example).

Would be good to check `invalid?` first in this method. Then it stops doing work if the coordinates are bad (doesn't run the horizontal/vertical/diagonal checks).